endpoints/openrtb2/auction.go

```diff
@@ -135,13 +125,20 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 	usersyncs := usersync.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie))
 	if req.App != nil {
 		labels.Source = pbsmetrics.DemandApp
-	} else {
+		labels.RType = pbsmetrics.ReqTypeORTB2App
+		if req.App.Publisher != nil && req.App.Publisher.ID != "" {
+			labels.PubID = req.App.Publisher.ID
+		}
+	} else { //req.Site != nil
 		labels.Source = pbsmetrics.DemandWeb
 		if usersyncs.LiveSyncCount() == 0 {
 			labels.CookieFlag = pbsmetrics.CookieFlagNo
 		} else {
 			labels.CookieFlag = pbsmetrics.CookieFlagYes
 		}
+		if req.Site.Publisher != nil && req.Site.Publisher.ID != "" {
```

Comment:
I think we should keep the `req.Site != nil` check in this if condition unless we are sure that it will always be non-nil in this flow of the code. Similar for video_auction too.

Reply:
In this instance we are because the check in line 303 of the `validateRequest` function, guarantees `req.Site` and `req.App` are not both NULL nor not NULL

Reply:
We are sure Mansi. The `parseRequest(r)` function gets executed in line 110 just before our if statements here this function in turn calls `validateRequest(r)` that makes the following check:
```
 296     if (req.Site == nil && req.App == nil) || (req.Site != nil && req.App != nil) {
 297         errL = append(errL, errors.New("request.site or request.app must be defined, but not both."))
 298         return errL
 299     }
 300
 301 +-- 19 lines: if err := deps.validateSite(req.Site); err != nil {-----------------------------------
 320
 321     return errL
 322 }
endpoints/openrtb2/auction.go [RO]
```